### ISSUE TITLE
Terminology in the local laplacian filter

### ISSUE THREAD

**Issue by martinuray** [halide/Halide#4550]
I came across the local Laplacian filter in the apps section and started digging a bit deeper into that filter by reading the papers below and trying the filter in [Matlab](https://de.mathworks.com/help/images/ref/locallapfilt.html).

The major differences I observed are:
a.) the first parameter of the filter is not a sigma (as used by matlab and stated in the paper). Instead a `levels` is used. Digging in the code, I found that the levels is the number of intensity levels.

b.) at least in the c++ implementation, instead of passing alpha as it is, `alpha = alpha / (levels -1)`, what makes the α much smaller.

What is the idea behind both implementations? I am very curious about that. :-)

----
Aubry, M., Paris, S., Hasinoff, S. W., Kautz, J., & Durand, F. (2014). Fast Local Laplacian Filters. ACM Transactions on Graphics, 33(5), 1–14. https://doi.org/10.1145/2629645
Paris, S., Hasinoff, S. W., & Kautz, J. (2015). Local Laplacian filters. Communications of the ACM, 58(3), 81–91. https://doi.org/10.1145/2723694

**Comment by abadams** [comment]
The implementation doesn't track the parameters of the original paper exactly (We're implementing Aubry's fast variant), but was rather based on me reading the paper and talking to Sylvain (Paris), and writing an implementation with parameters that made sense to me at the time. I think I set the standard deviation of the Gaussian using the number of levels instead of vice-versa because it's naturally quantized in Aubry's version anyway, so using the number of levels made more sense to me. I think I rescaled alpha to make the two parameters more orthogonal. Narrowing the Gaussian by increasing number of levels makes the slope of it steeper, so I wanted to scale down alpha at the same time to hold the slope (the maximum possible local increase in contrast) constant as a function of levels. This was 8 years ago now though, so my recollection is hazy.
